src/main/kotlin/com/pubnub/api/subscribe/eventengine/effect/HandshakeReconnectEffect.kt

```diff
@@ -0,0 +1,69 @@
+package com.pubnub.api.subscribe.eventengine.effect
+
+import com.pubnub.api.PubNubException
+import com.pubnub.api.endpoints.remoteaction.RemoteAction
+import com.pubnub.api.eventengine.EventQueue
+import com.pubnub.api.eventengine.ManagedEffect
+import com.pubnub.api.subscribe.eventengine.event.Event
+import com.pubnub.api.subscribe.eventengine.event.SubscriptionCursor
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.TimeUnit
+
+class HandshakeReconnectEffect(
+    private val remoteAction: RemoteAction<SubscriptionCursor>,
+    private val eventQueue: EventQueue,
+    private val policy: RetryPolicy,
+    private val executorService: ScheduledExecutorService,
+    private val handshakeReconnectInvocation: SubscribeEffectInvocation.HandshakeReconnect,
+) : ManagedEffect {
+
+    @Transient
+    private var scheduled: ScheduledFuture<*>? = null
+
+    @Transient
+    private var cancelled = false
+
+    @Synchronized
+    override fun runEffect() {
+        if (cancelled) { // is it ok to be here or should be after "if (delay == null)"
```

Comment:
Do we need this comment? Because it's synchronized the cancelled can't change during this call so it's better to bail out as quickly as possible

Reply:
ok, I will delete it.